dev/com.ibm.ws.microprofile.config_fat_tck/fat/src/org/eclipse/microprofile/config/tck/ConfigTckPackageTest.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.microprofile.config.tck;
+
+import java.io.File;
+import java.nio.file.Files;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import componenttest.annotation.AllowedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.MvnUtils;
+
+/**
+ * This is a test class that runs a whole Maven TCK as one test FAT test.
+ * There is a detailed output on specific
+ */
+@SuppressWarnings("restriction")
+@RunWith(FATRunner.class)
+public class ConfigTckPackageTest {
+
+    @Server("FATServer")
+    public static LibertyServer server;
+
+    // These are the jar names subsets that are devoid of specific version numbers
+    String[] jarsUsed = { "com.ibm.websphere.org.eclipse.microprofile.config",
```

Comment:
This is the 1.1 TCK so don't we need to specify the 1.1 version of the api and impl jars? At some point there will be other versions.

Reply:
I think we will when we come to having to co-existing versions. As discussed, my plan is to introduce something like regex support when this is needed - this is not completely straightforward as way it works in maven is the pom.xml can use a environment variable with ${variable}. So if I 1) get rid of the list of jar 'keys' by scanning the ${variables} in pom.xml systemPath elements and 2) have these as a way that can make some numbers significant and some not significant - unless they 'key' is duplicated - then these would have to be some sort of 'escaped' regex that are valid environment variables. Or some similar scheme.